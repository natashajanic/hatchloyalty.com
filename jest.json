{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
