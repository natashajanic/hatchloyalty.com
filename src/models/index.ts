import { FluidObject } from 'gatsby-image'

export interface IBlogPost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    author: string
    date: string
    featuredImage?: IGraphQLImage
    tags: string[]
    title: string
  }
  html: string
}

export interface IGraphQLImage {
  childImageSharp: {
    fluid: FluidObject
  }
}

export interface IRelease {
  changes: Array<{ component: string; description: string; link?: string }>
  date: string
  name: string
}

export interface IResourcePost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    title: string
  }
  html: string
}

export interface ITeamMember {
  headshot?: IGraphQLImage
  name: string
  position: string
}
