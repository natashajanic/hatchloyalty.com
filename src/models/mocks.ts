import { IRelease, IBlogPost, IResourcePost, ITeamMember } from '.'

export const mockBlogPost: IBlogPost = {
  excerpt: 'This is the excerpt for a blog post.',
  fields: {
    slug: '/2019-02-25-test-blog-post/',
  },
  frontmatter: {
    author: 'John Doe',
    date: 'February 25, 2019',
    featuredImage: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1,
          src: 'test/imgs/test-blog-post.png',
          srcSet: '',
          sizes: '',
        },
      },
    },
    tags: ['Test', 'Press Release'],
    title: 'Test Blog Post',
  },
  html: '<div>Test blog post body content.</div>',
}

export const mockRelease: IRelease = {
  changes: [
    { component: 'Test', description: 'Test Change 1' },
    {
      component: 'Test HQ',
      description: 'Test Change 2',
      link: 'https://developer.hatchloyalty.com/test/change-2',
    },
  ],
  date: 'January 2019',
  name: 'Test Release 2019.01',
}

export const mockResourcePost: IResourcePost = {
  excerpt: 'This is the excerpt for a resource post.',
  fields: {
    slug: '/test-resource-post/',
  },
  frontmatter: {
    title: 'Test Resource Post',
  },
  html: '<div>Test resource post body content.</div>',
}

export const mockTeamMember: ITeamMember = {
  headshot: {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        src: 'test/imgs/john-doe.png',
        srcSet: '',
        sizes: '',
      },
    },
  },
  name: 'John Doe',
  position: 'CEO',
}
