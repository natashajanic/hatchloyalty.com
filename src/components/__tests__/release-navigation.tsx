import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import ReleaseNavigation, {
  IReleaseNavigationProps,
} from '../release-navigation'
import { mockRelease } from 'src/models/mocks'

describe('ReleaseNavigation', () => {
  let props: IReleaseNavigationProps
  let mountedComponent: ShallowWrapper<any, any, ReleaseNavigation>

  beforeEach(() => {
    props = {
      next: { ...mockRelease, name: 'Release 009' },
      previous: { ...mockRelease, name: 'Release 007' },
    }
    mountedComponent = shallow(<ReleaseNavigation {...props} />)
  })

  describe('render', () => {
    it('renders a link to the previous release', async () => {
      expect(mountedComponent.find(Link).length).toEqual(3)
      const prevLink = mountedComponent.find(Link).at(0)
      expect(prevLink.prop('to')).toEqual('/releases/release-007')
      expect(prevLink.text()).toContain(props.previous!.name)
    })

    it('omits the link to the previous release if none is specified', async () => {
      mountedComponent.setProps({ previous: undefined })
      expect(mountedComponent.find(Link).length).toEqual(2)
    })

    it('renders a link to the full changelog', async () => {
      expect(mountedComponent.find(Link).length).toEqual(3)
      const changelogLink = mountedComponent.find(Link).at(1)
      expect(changelogLink.prop('to')).toEqual('/releases')
      expect(changelogLink.text()).toContain('View All Releases')
    })

    it('renders a link to the next release', async () => {
      expect(mountedComponent.find(Link).length).toEqual(3)
      const nextLink = mountedComponent.find(Link).at(2)
      expect(nextLink.prop('to')).toEqual('/releases/release-009')
      expect(nextLink.text()).toContain(props.next!.name)
    })

    it('omits the link to the next release if none is specified', async () => {
      mountedComponent.setProps({ next: undefined })
      expect(mountedComponent.find(Link).length).toEqual(2)
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })

    it('matches the expected snapshot when there is no previous release', async () => {
      mountedComponent.setProps({ previous: undefined })
      expect(mountedComponent).toMatchSnapshot()
    })

    it('matches the expected snapshot when there is no next release', async () => {
      mountedComponent.setProps({ next: undefined })
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
