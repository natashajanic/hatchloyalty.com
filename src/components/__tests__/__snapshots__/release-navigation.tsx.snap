// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseNavigation render matches the expected snapshot 1`] = `
<Styled(Component)
  blacklist={
    Array [
      "css",
      "fontSize",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "textAlign",
    ]
  }
  fontSize={3}
  my={3}
  textAlign="center"
>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    style={
      Object {
        "float": "left",
        "textAlign": "left",
      }
    }
    width={200}
  >
    <ForwardRef
      to="/releases/release-007"
    >
      Previous: 
      Release 007
    </ForwardRef>
     
  </Styled(Component)>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    display="inline-block"
  >
    <ForwardRef
      to="/releases"
    >
      View All Releases
    </ForwardRef>
  </Styled(Component)>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    style={
      Object {
        "float": "right",
        "textAlign": "right",
      }
    }
    width={200}
  >
     
    <ForwardRef
      to="/releases/release-009"
    >
      Next: 
      Release 009
    </ForwardRef>
  </Styled(Component)>
</Styled(Component)>
`;

exports[`ReleaseNavigation render matches the expected snapshot when there is no next release 1`] = `
<Styled(Component)
  blacklist={
    Array [
      "css",
      "fontSize",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "textAlign",
    ]
  }
  fontSize={3}
  my={3}
  textAlign="center"
>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    style={
      Object {
        "float": "left",
        "textAlign": "left",
      }
    }
    width={200}
  >
    <ForwardRef
      to="/releases/release-007"
    >
      Previous: 
      Release 007
    </ForwardRef>
     
  </Styled(Component)>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    display="inline-block"
  >
    <ForwardRef
      to="/releases"
    >
      View All Releases
    </ForwardRef>
  </Styled(Component)>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    style={
      Object {
        "float": "right",
        "textAlign": "right",
      }
    }
    width={200}
  >
     
  </Styled(Component)>
</Styled(Component)>
`;

exports[`ReleaseNavigation render matches the expected snapshot when there is no previous release 1`] = `
<Styled(Component)
  blacklist={
    Array [
      "css",
      "fontSize",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "textAlign",
    ]
  }
  fontSize={3}
  my={3}
  textAlign="center"
>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    style={
      Object {
        "float": "left",
        "textAlign": "left",
      }
    }
    width={200}
  >
     
  </Styled(Component)>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    display="inline-block"
  >
    <ForwardRef
      to="/releases"
    >
      View All Releases
    </ForwardRef>
  </Styled(Component)>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    style={
      Object {
        "float": "right",
        "textAlign": "right",
      }
    }
    width={200}
  >
     
    <ForwardRef
      to="/releases/release-009"
    >
      Next: 
      Release 009
    </ForwardRef>
  </Styled(Component)>
</Styled(Component)>
`;
